src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FootnoteService.cs

```diff
@@ -60,83 +56,70 @@ public class FootnoteService : IFootnoteService
                 .OnSuccess(async footnote => await GetFootnote(releaseId, footnote.Id)
                 );
         }
 
         public async Task<Either<ActionResult, List<Footnote>>> CopyFootnotes(Guid sourceReleaseId, Guid destinationReleaseId)
         {
-            return await GetFootnotes(sourceReleaseId)
-                .OnSuccess(async footnotes => 
-                    await footnotes.ForEachAsync(async footnote =>
-                    {
-                        var filters = footnote.Filters
-                            .Select(filterFootnote => filterFootnote.FilterId).ToList();
-                        var filterGroups = footnote.FilterGroups
-                            .Select(filterGroupFootnote => filterGroupFootnote.FilterGroupId).ToList();
-                        var filterItems = footnote.FilterItems
-                            .Select(filterItemFootnote => filterItemFootnote.FilterItemId).ToList();
-                        var indicators = footnote.Indicators
-                            .Select(indicatorFootnote => indicatorFootnote.IndicatorId).ToList();
-                        var subjects = footnote.Subjects
-                            .Select(subjectFootnote => subjectFootnote.SubjectId).ToList();
-
-                        return await CreateFootnote(destinationReleaseId,
-                            footnote.Content,
-                            filters,
-                            filterGroups,
-                            filterItems,
-                            indicators,
-                            subjects);
-                    }));
+            return await _contentPersistenceHelper.CheckEntityExists<Release>(destinationReleaseId)
+                .OnSuccessDo(_userService.CheckCanUpdateRelease)
+                .OnSuccess(() => GetFootnotes(sourceReleaseId))
+                .OnSuccess(async footnotes =>
+                {
+                    return await footnotes
+                        .ToAsyncEnumerable()
+                        .SelectAwait(async footnote =>
+                        {
+                            var filters = footnote.Filters
+                                .Select(filterFootnote => filterFootnote.FilterId).ToList();
+                            var filterGroups = footnote.FilterGroups
+                                .Select(filterGroupFootnote => filterGroupFootnote.FilterGroupId).ToList();
+                            var filterItems = footnote.FilterItems
+                                .Select(filterItemFootnote => filterItemFootnote.FilterItemId).ToList();
+                            var indicators = footnote.Indicators
+                                .Select(indicatorFootnote => indicatorFootnote.IndicatorId).ToList();
+                            var subjects = footnote.Subjects
+                                .Select(subjectFootnote => subjectFootnote.SubjectId).ToList();
```

Comment:
Not a biggy, but could suffix these with `Ids` e.g. `filterIds` to name them more appropriately.

Reply:
Made this change now as part of the new commit altering from lists to sets.